wallets/react/src/useProviders.ts

```diff
@@ -0,0 +1,120 @@
+  ProviderProps,
+  Providers,
+} from './index.js';
+import type { ConnectResult } from './legacy/mod.js';
+import type { LegacyState } from '@rango-dev/wallets-core/legacy';
+import type { SignerFactory } from 'rango-types';
+
+import {
+  findProviderByType,
+  separateLegacyAndHubProviders,
+  useHubAdapter,
+} from './hub/mod.js';
+import { useLegacyProviders } from './legacy/mod.js';
+
+/*
+ * We have two separate interface for our providers: legacy and hub.
+ * This hook sits between this two interface by keeping old interface as main API and try to add Hub providers by using an adapter.
+ * For gradual migrating and backward compatibility, we are supporting hub by an adapter besides of the old one.
+ */
+function useProviders(props: ProviderProps) {
+  const { providers, ...restProps } = props;
+  const [legacyProviders, hubProviders] = separateLegacyAndHubProviders(
+    providers,
+    {
+      isExperimentalEnabled: restProps.configs?.isExperimentalEnabled,
+    }
+  );
+
+  const legacyApi = useLegacyProviders({
+    ...restProps,
+    providers: legacyProviders,
+  });
+  const hubApi = useHubAdapter({
+    ...restProps,
+    providers: hubProviders,
+    allVersionedProviders: providers,
+  });
+
+  const api: ProviderContext = {
+    canSwitchNetworkTo(type, network): boolean {
+      if (findProviderByType(hubProviders, type)) {
+        return hubApi.canSwitchNetworkTo(type, network);
+      }
+      return legacyApi.canSwitchNetworkTo(type, network);
+    },
+    async connect(type, network): Promise<ConnectResult[]> {
+      const hubProvider = findProviderByType(hubProviders, type);
+      if (hubProvider) {
+        return await hubApi.connect(type, network);
+      }
+
+      return await legacyApi.connect(type, network);
+    },
+    async disconnect(type): Promise<void> {
+      const hubProvider = findProviderByType(hubProviders, type);
+      if (hubProvider) {
+        return await hubApi.disconnect(type);
+      }
+
+      return await legacyApi.disconnect(type);
+    },
+    async disconnectAll() {
+      return await Promise.allSettled([
+        hubApi.disconnectAll(),
+        legacyApi.disconnectAll(),
+      ]);
+    },
+    async getSigners(type): Promise<SignerFactory> {
+      const hubProvider = findProviderByType(hubProviders, type);
+      if (hubProvider) {
+        return hubApi.getSigners(type);
+      }
+      return legacyApi.getSigners(type);
+    },
+    getWalletInfo(type): ExtendedWalletInfo {
+      const hubProvider = findProviderByType(hubProviders, type);
+      if (hubProvider) {
+        return hubApi.getWalletInfo(type);
+      }
+
+      return legacyApi.getWalletInfo(type);
+    },
+    providers(): Providers {
+      let output: Providers = {};
+      if (hubProviders.length > 0) {
+        output = { ...output, ...hubApi.providers() };
+      }
+      if (legacyProviders.length > 0) {
+        output = { ...output, ...legacyApi.providers() };
+      }
+
+      return output;
```

Comment:
What is the problem with this?
```suggestion
      return { ...hubApi.providers(), ...legacyApi.providers() };
```

Reply:
Not sure. Only god knows.
updated.

Reply:
OK. I can officially say god is not alone anymore.
After checking more, when `isExperimentalEnabled` set to false, `separateLegacyAndHubProviders` will return only legacy providers.
Your code causes hub providers to be returned always, but when it's disabled it we shouldn't run `...hubApi.providers()`.

So I revert your suggestion and kept the original code.